### PR TITLE
fix(lane_change): update target lane obj block condition

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Originally, if there was a stationary vehicle in the target lane, the ego stopped without taking a lane change margin. However, it should take the margin in some cases, like in the video.

I modified the condition for not taking a lane change margin from "when there's a stationary vehicle in the target lane" to "when there's a stationary vehicle in the target lane within the lane change zone".

[Screencast from 2023年10月13日 00時44分25秒.webm](https://github.com/autowarefoundation/autoware.universe/assets/21360593/848fffe5-623c-410f-a454-46e85f0fc57f)

## Related links

None

## Tests performed

Run psim

## Notes for reviewers

It is easy to review commits one by one.

## Interface changes

None

## Effects on system behavior

improve lane change performance

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by TakaHoribe** [comment]
TODO:

lane change stop line is here
image


However, sometimes `stopping_distance_for_obj` is shorter than `distance_to_target_lane_obj`. Why...?
image

File: planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp [hunk 1]
```diff
@@ -206,67 +206,79 @@ void NormalLaneChange::insertStopPoint(
...
+    const bool has_blocking_target_lane_obj = std::any_of(
+      target_objects.target_lane.begin(), target_objects.target_lane.end(), [&](const auto & o) {
+        const auto v = std::abs(o.initial_twist.twist.linear.x);
+        if (v > lane_change_parameters_->stop_velocity_threshold) {
+          return false;
+        }
+        const double distance_to_target_lane_obj = utils::getSignedDistance(
+          path.points.front().point.pose, o.initial_pose.pose, status_.target_lanes);
+        return stopping_distance_for_obj < distance_to_target_lane_obj &&
+               distance_to_target_lane_obj < distance_to_ego_lane_obj;
```
[the remark this answers is not in the document]
Comment (kosuke55) [reply planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp:276]: `rss_dist` may be minus?

[the remark this answers is not in the document]
Comment (kosuke55) [reply planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp:276]: sorry this is my misunderstading

File: planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp [hunk 2]
```diff
@@ -206,67 +206,79 @@ void NormalLaneChange::insertStopPoint(
...
+                                           stop_point_buffer - rss_dist -
+                                           getCommonParam().base_link2front;
+
+    //  If the target lane in the lane change section is blocked by a stationary obstacle, there
+    //  is no reason for stopping with a lane change margin. Instead, stop right behind the
+    //  obstacle.
+    //  ----------------------------------------------------------
+    //                            [obj]>
+    //  ----------------------------------------------------------
+    //    [ego]>          | <--- lane change margin --->  [obj]>
```
[the remark this answers is not in the document]
Comment (TakaHoribe) [reply planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp:265]: Not same. 
- `stopping_distance_for_obj` is the distance from path front to object
- `lane change margin` is the distance from stop point to object

**Pull request merged**